### PR TITLE
Support MN mode for evaluators

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Comment by yuyu2172** [comment]
Change tests:
For MN ones, only check consistency with single node.

**Comment by yuyu2172** [comment]
pfnCI, test this please

File: tests/extensions_tests/evaluator_tests/test_detection_coco_evaluator.py [hunk 1]
```diff
@@ -104,4 +107,75 @@ def test_current_report(self):
...
+        areas = [[np.array([(bb[2] - bb[0]) * bb[3] - bb[0]])
+                  for bb in bbox] for bbox in bboxes]
+        crowdeds = [np.zeros((5,)) for _ in range(10)]
+        self.dataset = TupleDataset(
+            np.random.uniform(size=(10, 3, 32, 48)),
+            bboxes, labels, areas, crowdeds)
+        self.initial_count = initial_count
+        self.batchsize = batchsize
+
+    @attr.mpi
```
Comment (Hakuyume) [inline tests/extensions_tests/evaluator_tests/test_detection_coco_evaluator.py]: You can put `@attr.mpi` on the class.
https://github.com/chainer/chainercv/blob/master/tests/links_tests/model_tests/ssd_tests/test_multibox_loss.py#L126-L127

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Pull request merged**